src/main/java/org/opensearch/neuralsearch/search/util/HybridQueryScoreDocsMerger.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.util;
+
+import lombok.NoArgsConstructor;
+import org.apache.lucene.search.ScoreDoc;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+import static org.opensearch.neuralsearch.search.util.HybridSearchResultFormatUtil.isHybridQueryScoreDocElement;
+
+/**
+ * Merges two ScoreDoc arrays into one
+ */
+@NoArgsConstructor
+public class HybridQueryScoreDocsMerger<T extends ScoreDoc> {
```

Comment:
lets make this class package private, as the functions inside of this class should be limited to HybridQuery clause only

Reply:
ack

Reply:
pushed commit, I have to move both merger classes to `query` package otherwise package private doesn't work.